MapleServer2/Managers/GlobalEventManager.cs

```diff
@@ -30,4 +32,19 @@ public GlobalEvent GetEventById(int id)
     {
         return GlobalEventList.GetValueOrDefault(id);
     }
+
+    public GlobalEvent GetCurrentEvent()
+    {
+        return GlobalEventList.Values.FirstOrDefault();
```

Comment:
It doesn't look like the current event is guaranteed to be the first one to me, if someone schedules an event which is going to run in 10 hours, then schedules another to run in 1 hour, the second will run first.

Reply:
That's the point, isn't it? You want to get the current event. 
A scheduled event doesn't add the global event into the event manager until the scheduled time triggers.

Reply:
My mistake, I hadn't seen that `EventCommand` no longer calls `AddEvent` immediately.